### DISCUSSION TITLE
Exporter softflowd

### DISCUSSION THREAD

**Discussion by baliushko** [akvorado/akvorado#225] [Q&A]
I have linux server with softflowd
`Linux border-Stella 4.19.0-20-amd64 #1 SMP Debian 4.19.235-1 (2022-03-17) x86_64 GNU/Linux`
`softflowd version 0.9.9.`
`NET-SNMP version:  5.7.3`
`/usr/sbin/softflowd -i bond0.2776 -p /run/softflowd/bond0.2776.pid -c /run/softflowd/bond0.2776.ctl -n 172.16.0.176:2055 -v 9`
172.16.0.176 this is akvorado server

akvorado server has snmp access to linux server
![image](https://user-images.githubusercontent.com/110233899/198229845-8da796e5-850b-4c35-8cd9-8461340f43bb.png)

I see the flow but the exporter is not being created.
![image](https://user-images.githubusercontent.com/110233899/198230408-f7430aed-66a8-4fe2-875e-d21361a06d92.png)

![image](https://user-images.githubusercontent.com/110233899/198231852-2cda81b3-12fc-4769-bf74-dbeafb343547.png)

There is also a flow from junos. He works.
![image](https://user-images.githubusercontent.com/110233899/198232343-dba17920-c1f9-4c73-b716-7a6a14fc337a.png)

**Comment by vincentbernat** [comment]
Check the troubleshooting section of the documentation and the counters you have for the non-working exporter.

**Comment by baliushko** [comment]
```

akvorado_inlet_core_flows_errors{error="input and output interfaces missing",exporter="172.16.0.194"} 3.2643756e+08
akvorado_inlet_flow_decoder_netflow_errors_count{error="error decoding",exporter="172.16.0.194"} 5.785294e+06

```

**Reply by vincentbernat** [reply]
Use tcpdump to check what the flows look like. It seems the interfaces are set to 0.

**Comment by baliushko** [comment]
Softflowd did not send interface indexes. I fixed it.
![image](https://user-images.githubusercontent.com/110233899/198535802-a8a2049b-c4e1-4efa-af82-b5a51f17f183.png)

But now a new problem, srcAS and dstAS is not displayed.
![image](https://user-images.githubusercontent.com/110233899/198536446-15e60ab8-1553-456f-86a2-e29e180d52b4.png)

**Reply by vincentbernat** [reply]
It requires either to be embedded in sflow (and I suppose it is not) or you need to configure GeoIP.

**Reply by baliushko** [reply]
I downloaded the bases and put them on the path from the file and restarted the docker.
![image](https://user-images.githubusercontent.com/110233899/198544995-046703a5-2619-42d8-b549-9081232eed9c.png)

This did not help, how can I find out if I configured the geoip correctly

**Reply by baliushko** [reply]
akvorado-geoip-1  exited (1)
![image](https://user-images.githubusercontent.com/110233899/198547711-a41cf16a-4e55-43e8-99c3-6aa52cdf5a80.png)
I dont find logs
![image](https://user-images.githubusercontent.com/110233899/198547979-07b93685-e2cd-48b3-a6f7-8fbefc968689.png)

**Reply by baliushko** [reply]
Destination country is not working =(
![image](https://user-images.githubusercontent.com/110233899/198550522-c3c65533-5336-401d-89c9-d2ff14bc3908.png)

**Reply by baliushko** [reply]
![image](https://user-images.githubusercontent.com/110233899/198552624-f6b9aa1b-1d8a-49b8-b4eb-c3b7bd491d02.png)

I do not find in the documentation where to define these variables.

**Reply by baliushko** [reply]
I have defined variables in docker-compose.yml. Everything worked, thanks.

**Reply by wouterprins** [reply]
> Softflowd did not send interface indexes. I fixed it. ![image](https://user-images.githubusercontent.com/110233899/198535802-a8a2049b-c4e1-4efa-af82-b5a51f17f183.png)

Hi, also a softflowd (0.9.9) user here.. Happy to see you got it working and thanks for this hint regarding the ifindex. Im using the following configuration on OpenBSD in /etc/rc.conf.local which is similar like your setup:

softflowd_flags="-i 20:vlan822 -n akvoradoip:2055 -v9"

Running into the following errors on akvorado (i've tried using -s <sample-rate>) but the sample rate error does not disappear. Using softflowctl send-template does not help to fix the "template not found" message.

akvorado_inlet_core_flows_errors{error="sampling rate missing",exporter="<sourceip>"} 219873
akvorado_inlet_flow_decoder_netflow_errors_count{error="template not found",exporter="source-ip"} 4

Did you encounter this as well?

**Reply by nneo16** [reply]
For who using the softflowd openwrt, config file /etc/config/softflowd, adding the index before interface name, should be like this:

from
> config softflowd
        option enabled '1'
        option interface 'br-lan'

to
> config softflowd
        option enabled '1'
        option interface '0:br-lan'
